File: pip.py

user_preferences = [{}]

def add_user(user_id, preferred_categories):
    user_preferences[user_id] = preferred_categories

def get_recommendations(user_id):
    preferred_categories = user_preferences.get(user_id, [])
    if not preferred_categories:
        return []
    
    return ["All items in our categories"]

def get_recommendations(user_id):
    preferred_categories = user_preferences.get(user_id, [])
    if not preferred_categories:
        return []
    
    return ["Recommended Items"]

File: pip.js


const userPreferences = {};

function addUser(userId, preferredCategories) {
    userPreferences[userId] = preferredCategories;
    return userPreferences
}
let addedUser = addUser("001", ["trousers", "shorts", "bag"])
console.log({addedUser});

function getRecommendations(userId) {
    const preferredCategories = userPreferences[userId];
    if (!preferredCategories) return [];
    
    return ["Recommendation"];
}

console.log(getRecommendations("001"));
